quinn/transformations.py

```diff
@@ -87,28 +88,193 @@ def to_snake_case(s: str) -> str:
     ascending and descending order, respectively. If any other value is provided for
     the ``sort_order`` parameter, a ``ValueError`` will be raised.
 
     :param df: A DataFrame
-    :type df: pandas.DataFrame
+    :type df: pyspark.sql.DataFrame
     :param sort_order: The order in which to sort the columns in the DataFrame
     :type sort_order: str
+    :param sort_nested: Whether to sort nested structs or not. Defaults to false.
+    :type sort_nested: bool
     :return: A DataFrame with the columns sorted in the chosen order
-    :rtype: pandas.DataFrame
+    :rtype: pyspark.sql.DataFrame
     """
-    sorted_col_names = None
-    if sort_order == "asc":
-        sorted_col_names = sorted(df.columns)
-    elif sort_order == "desc":
-        sorted_col_names = sorted(df.columns, reverse=True)
-    else:
-        raise ValueError(
-            "['asc', 'desc'] are the only valid sort orders and you entered a sort order of '{sort_order}'".format(
-                sort_order=sort_order
+
+    def parse_sort_order(sort_order: str) -> bool:
+        if sort_order not in ["asc", "desc"]:
+            raise ValueError(
+                "['asc', 'desc'] are the only valid sort orders and you entered a sort order of '{sort_order}'".format(
+                    sort_order=sort_order
+                )
             )
-        )
-    return df.select(*sorted_col_names)
+        reverse_lookup = {
+            "asc": False,
+            "desc": True,
+        }
+        return reverse_lookup[sort_order]
+
+    def sort_top_level_cols(df, is_reversed) -> DataFrame:
+        # sort top level columns
+        sorted_col_names = sorted(df.columns, reverse=is_reversed)
+        return df.select(*sorted_col_names)
+
+    def sort_nested_cols(schema, is_reversed, base_field="") -> list[str]:
+        # recursively check nested fields and sort them
+        # https://stackoverflow.com/questions/57821538/how-to-sort-columns-of-nested-structs-alphabetically-in-pyspark
+        # Credits: @pault for logic
+
+        def parse_fields(
+            fields_to_sort: list, parent_struct, is_reversed: bool
+        ) -> list:
+            sorted_fields: list = sorted(
+                fields_to_sort,
+                key=lambda x: x["name"],
+                reverse=is_reversed,
+            )
+
+            results = []
+            for field in sorted_fields:
+                new_struct = StructType([StructField.fromJson(field)])
+                new_base_field = parent_struct.name
+                if base_field:
+                    new_base_field = base_field + "." + new_base_field
+
+                results.extend(
+                    sort_nested_cols(new_struct, is_reversed, base_field=new_base_field)
+                )
+            return results
+
+        def handle_array_type(parent_struct: StructField, is_reversed: bool) -> str:
+            def format_array_selection(
```

Comment:
Cognitive complexity of this part is very high. It is 4th level of nested functions. And it is called once. Do we really need it? Cause code is very hard to read